netsync/manager.go

```diff
@@ -454,9 +454,10 @@ func (sm *SyncManager) handleCheckSyncPeer() {
 	// Update network stats at the end of this tick.
 	defer sm.syncPeerState.updateNetwork(sm.syncPeer)
 
-	// Check network speed of the sync peer and its last block time.
-	if (sm.syncPeerState.validNetworkSpeed(sm.minSyncPeerNetworkSpeed) < maxNetworkViolations) &&
-		(time.Since(sm.syncPeerState.lastBlockTime) <= maxLastBlockTime) {
+	// Check network speed of the sync peer and its last block time. If we're currently
+	// flushing the cache skip this round.
+	if ((sm.syncPeerState.validNetworkSpeed(sm.minSyncPeerNetworkSpeed) < maxNetworkViolations) &&
+		(time.Since(sm.syncPeerState.lastBlockTime) <= maxLastBlockTime)) || sm.chain.UtxoCacheFlushInProgress() {
```

Comment:
I would move this check for the UtxoCacheFlush down below. For now I would just add it to the block height check. If there is a flush in progress just set the lastBlockTime and reset the violations. Worst case the recovery happens a little slower. If you do the check here, immediately after flush it is very possible we will pick a new peer.

Reply:
Also if there is a lot of date to flush, then it is obvious the sync is happening, so I think this makes logical sense as well.